packages/components/pie-spinner/src/spinner.scss

```diff
@@ -1 +1,88 @@
 @use '@justeattakeaway/pie-css/scss' as p;
+
+*,
+*:before,
+*:after {
+    box-sizing: border-box;
+}
+
+/**
+* Shorthand mixin for updating the HSL custom properties for the loading spinner.
+* Takes in the name of the color to be split into its HSL components.
+*/
+@mixin spinner-base-colors($color) {
+    --spinner-base-color-h: var(#{$color}-h);
+    --spinner-base-color-s: var(#{$color}-s);
+    --spinner-base-color-l: var(#{$color}-l);
+}
+
+// Spin animation for loading state
+@keyframes rotate360 {
+    from {
+        transform: rotate(0deg);
+    }
+
+    to {
+        transform: rotate(360deg);
+    }
+}
+
+.c-spinner {
+    --spinner-size: 24px;
+    --spinner-left-color: hsl(var(--spinner-base-color-h), var(--spinner-base-color-s), var(--spinner-base-color-l), 1);
+    --spinner-right-color: hsl(var(--spinner-base-color-h), var(--spinner-base-color-s), var(--spinner-base-color-l), 0.35);
+
+    block-size: var(--spinner-size);
+    inline-size: var(--spinner-size);
+    border-radius: var(--dt-radius-rounded-e);
+    border-width: calc(var(--spinner-size) / 8);
+    border-style: solid;
+    border-color: var(--spinner-left-color) var(--spinner-right-color) var(--spinner-right-color) var(--spinner-left-color);
+    will-change: transform;
+    animation: rotate360 1.15s linear infinite;
+
+    @include spinner-base-colors('--dt-color-content-brand');
+
+    &[variant='brand'] {
+        /* Same as default styles */
+    }
+
+    &[variant='secondary'] {
+        @include spinner-base-colors('--dt-color-content-interactive-secondary');
+    }
+
+    &[variant='inverse'] {
+        @include spinner-base-colors('--dt-color-content-inverse');
+    }
+
+    &[size='xs'] {
+        --spinner-size: 16px;
+    }
+
+    &[size='s'] {
+        --spinner-size: 20px;
+    }
+
+    &[size='m'] {
+        /* Same as default styles */
+    }
+
+    &[size='l'] {
+        --spinner-size: 32px;
+    }
+
+    &[size='xl'] {
+        --spinner-size: 48px;
+    }
+}
+
+// The label is only required for screen readers so we need to visually hide it
+.c-spinner-label {
```

Comment:
Can you add a `// TODO` here to move this to a utility mixin in `pie-css`? (it would be good to make a ticket for this too)

Reply:
on my radar too 🤝 
thanks for flagging